SPECS/dracut/20overlayfs/overlayfs-mount.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/sh
+
+set -ex
+
+parse_cmdline_args() {
+    # Ensure that the 'dracut-lib' is present and loaded.
+    type getarg >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+    VERITY_MOUNT="mnt/verity_mnt"
+    OVERLAY_MOUNT="/mnt/overlay_mnt"
+    OVERLAY_MNT_OPTS="rw,nodev,nosuid,nouser,noexec"
+
+    # Retrieve the verity root. It is expected to be predefined by the dracut cmdline module.
+    [ -z "$root" ] && root=$(getarg root=)
+    # Check if the root variable matches the expected path.
+    # The path "/dev/mapper/root" is hardcoded here because it is a fixed target name
+    # generated by systemd-veritysetup. The name of this dm-verity target is determined
+    # by systemd and cannot be changed, hence the explicit check against this specific path.
+    [[ "$root" == *"/dev/mapper/root"* ]] || die "root mismatches."
+
+    # Retrieve the OverlayFS parameters.
+    [ -z "${overlayfs}" ] && overlayfs=$(getarg rd.overlayfs=)
+    # Retrieve the persistent volume for the OverlayFS.
+    [ -z "${overlayfs_persistent_volume}" ] && overlayfs_persistent_volume=$(getarg rd.overlayfs_persistent_volume=)
+}
+
+# Modified function to mount the physical partition
+mount_physical_partition() {
+    mkdir -p "${OVERLAY_MOUNT}"
+    # Leverage the partition from cmdline
+    local partition="${overlayfs_persistent_volume}"
+
+    if [ -z "${partition}" ]; then
+        # Fallback to volatile overlay if no persistent volume is specified
+        info "No overlayfs persistent volume specified. Creating a volatile overlay."
+        mount -t tmpfs tmpfs -o ${OVERLAY_MNT_OPTS} "${OVERLAY_MOUNT}" || \
+            die "Failed to create overlay tmpfs at ${OVERLAY_MOUNT}"
```

Comment:
`die` seems to be a perl specific function which would not work in `sh` unless we explicitly define a function.

Reply:
dracut defines bash common functions that are used across dracut modules.

Reply:
> dracut defines bash common functions that are used across dracut modules. 

TiL thanks